server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -0,0 +1,111 @@
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.remote.RemoteRefreshSegmentTracker;
+
+import java.io.IOException;
+
+/**
+ * Encapsulates all remote store stats
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreStats implements Writeable, ToXContentFragment {
+
+    private final RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats;
+
+    public RemoteStoreStats(RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats) {
+        this.remoteSegmentUploadShardStats = remoteSegmentUploadShardStats;
+    }
+
+    public RemoteStoreStats(StreamInput in) throws IOException {
+        remoteSegmentUploadShardStats = in.readOptionalWriteable(RemoteRefreshSegmentTracker.Stats::new);
+    }
+
+    public RemoteRefreshSegmentTracker.Stats getStats() {
+        return remoteSegmentUploadShardStats;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject()
+            .field("shardId", remoteSegmentUploadShardStats.shardId)
+            .field("latest_upload_files_count", remoteSegmentUploadShardStats.latestUploadFilesCount)
+            .field("latest_local_files_count", remoteSegmentUploadShardStats.latestLocalFilesCount)
+            .field("local_refresh_time", remoteSegmentUploadShardStats.localRefreshTime)
+            .field("local_refresh_seq_no", remoteSegmentUploadShardStats.localRefreshSeqNo)
+            .field("remote_refresh_time", remoteSegmentUploadShardStats.remoteRefreshTime)
+            .field("remote_refresh_seqno", remoteSegmentUploadShardStats.remoteRefreshSeqNo)
+            .field("bytes_lag", remoteSegmentUploadShardStats.bytesLag)
+            .field("inflight_upload_bytes", remoteSegmentUploadShardStats.inflightUploadBytes)
+            .field("inflight_uploads", remoteSegmentUploadShardStats.inflightUploads)
+            .field("rejection_count", remoteSegmentUploadShardStats.rejectionCount)
+            .field("consecutive_failure_count", remoteSegmentUploadShardStats.consecutiveFailuresCount);
+        builder.startObject("last_upload_time").field("started", -1).field("succeeded", -1).field("failed", -1);
+        builder.endObject();
+        builder.startObject("upload_bytes");
+        builder.field("started", remoteSegmentUploadShardStats.uploadBytesStarted)
+            .field("succeeded", remoteSegmentUploadShardStats.uploadBytesSucceeded)
+            .field("failed", remoteSegmentUploadShardStats.uploadBytesFailed);
+        builder.startObject("moving_avg");
+        builder.field("started", remoteSegmentUploadShardStats.uploadBytesMovingAverage)
+            .field("succeeded", remoteSegmentUploadShardStats.uploadBytesMovingAverage)
+            .field("failed", -1);
+        builder.endObject();
+        builder.startObject("per_sec_moving_avg");
```

Comment:
What is the difference between `moving_avg` and `per_sec_moving_avg`?

Reply:
this is a bit confusing, let me see if we can update the keys

upload_bytes.moving_avg is total size in bytes of segment files uploaded over last N uploads
upload_bytes.per_sec_moving_avg is bytes/sec(speed) of segment files upload over last N uploads